From b8423286b71611f6a01bc31c0420be3146a89b1b Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:38:39 -0400
Subject: [virt] kvm/x86: Sync DR7 on KVM_SET_DEBUGREGS

Message-id: <1406040016-3289-125-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86203
O-Subject: [RHEL7 PATCH v2 124/221] KVM: x86: Sync DR7 on KVM_SET_DEBUGREGS
Bugzilla: 1079841 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jan Kiszka <jan.kiszka@siemens.com>

Whenever we change arch.dr7, we also have to call kvm_update_dr7. In
case guest debugging is off, this will synchronize the new state into
hardware.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 9926c9fdbdd54bb229fe6fdbd15ca3af2b8425ae)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index 0e38bd5..dbe0f11 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -3010,6 +3010,7 @@ static int kvm_vcpu_ioctl_x86_set_debugregs(struct kvm_vcpu *vcpu,
  memcpy(vcpu->arch.db, dbgregs->db, sizeof(vcpu->arch.db));
  vcpu->arch.dr6 = dbgregs->dr6;
  vcpu->arch.dr7 = dbgregs->dr7;
+ kvm_update_dr7(vcpu);
 
  return 0;
 }
-- 
1.7.1